ouroboros-network/src/Ouroboros/Network/Block.hs

```diff
@@ -114,21 +119,34 @@ castHash (BlockHash b) = BlockHash b
 -- as a check, or in some contexts it disambiguates blocks from different forks
 -- that were in the same slot.
 --
-data Point block = Point {
-       pointSlot :: SlotNo,
-       pointHash :: ChainHash block
-     }
-   deriving (Eq, Ord, Show)
+-- It's a newtype rather than a type synonym, because using a type synonym
+-- would lead to ambiguity, since HeaderHash is a non-injective type family.
+newtype Point block = Point
+    { getPoint :: WithOrigin (Point.Block SlotNo (HeaderHash block))
```

Comment:
Aside from the name, what did you have in mind for this type synonym? I don't see how it could get much simpler than `Point.Block slot hash`.

Reply:
Hmm, what if in `Ouroboros.Consensus.Block` (exists already) we export a version of `Point` that is instantiated to `SlotNo` and `HeaderHash block`. It could just reuse the existing names like `Point`, `pointSlot`, `pointHash`, etc.
I'm not sure whether this will work, maybe it will conflict in modules that need both versions (a qualified import might help there?)

Reply:
Independent from my `Ouroboros.Consensus.Block` suggestion: IMO, something like `BlockPoint block` or `AtPoint block` is much simpler than `Point.Block SlotNo (HeaderHash block)`.

Reply:
How about this?

```Haskell
pattern GenesisPoint :: Point block
pattern GenesisPoint = Point Origin

pattern BlockPoint :: SlotNo -> HeaderHash block -> Point block
pattern BlockPoint { atSlot, withHash } = Point (At (Point.Block atSlot withHash))
```

Reply:
That works for me 👍